### PR TITLE
Warnings Bar Zoom Updates

### PR DESCRIPTION
**Pull request by OliverEGreen**
### Purpose

This PR updates the styling for warnings and errors displaying on nodes by introducing a new 12px bar at its base (which expands/collapses while maintaining a consistent gap with the node output preview). It also has added logic for handling changes to the workspace's zoom level, enabling nodes to display warning/error styling differently depending on the zoom level.

1mfgv51Rjf

Please Note: This PR does *not* disable click events for ports/tooltips etc at different zoom levels. These will be coordinated with Autodesk via a follow-up PR. It also doesn't include tests yet... because I'm going on holiday :) bye!

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

Updates node warning/error display styling.

### Reviewers

@SHKnudsen

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs [hunk 1]
```diff
@@ -349,6 +350,56 @@ public bool IsVisible
...
+        /// </summary>
+        [JsonIgnore]
+        public bool NodeOverlayVisible => IsFrozen ||
+                                          ErrorBubble?.NodeErrorsToDisplay.Count > 0 && WorkspaceViewModel.Zoom < WarningsBarVisibilityThreshold ||
+                                          ErrorBubble?.NodeWarningsToDisplay.Count > 0 && WorkspaceViewModel.Zoom < WarningsBarVisibilityThreshold;
+
+        /// <summary>
+        /// The color of the warning bar: orange for warnings, red for errors.
+        /// </summary>
+        public SolidColorBrush WarningBarColor
```
Comment (OliverEGreen) [inline src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs:377]: I'd love this to be a simple setter but I got some ugly threading issues here, so I fire this method in the existing dispatcher call.

File: src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs [hunk 2]
```diff
@@ -349,6 +350,56 @@ public bool IsVisible
...
         }
 
+        private const double WarningsBarVisibilityThreshold = 0.4;
+
+        /// <summary>
+        /// Determines whether or not the opaque warning bar is visible under nodes with
+        /// undismissed warning/error messages.
+        /// </summary>
+        [JsonIgnore]
+        public bool IsWarningBarVisible => WorkspaceViewModel.Zoom >= WarningsBarVisibilityThreshold &&
```
Comment (SHKnudsen) [inline src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs:360]: Wondering if all of this actually needs to be here on the viewModel, could we not move it all to the view and use converters?

So essentially we would just need to bind to the `ElementState` and then based on that and the Zoom value we determine wheter or not to display and what the color we need to show?

File: src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs (hunk 1 above)
Comment (SHKnudsen) [reply src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs:377]: If we need the setter we should probably make it `internal`

File: src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs (hunk 2 above)
Comment (SHKnudsen) [reply src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs:360]: There already is a `ZoomToVisibilityConverter` https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCoreWpf/UI/Converters.cs#L1550

If that controls the visibility of the UIElement and then we introduce a new converter like `ElementStateToBackgroundConverter` which controls the background color of the UIElement

Comment (SHKnudsen) [reply src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs:360]: the `ZoomToVisibilityConverter` isnt used anywhere anymore, but you can look at previous branches to see how it was used

File: src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs (hunk 1 above)
Comment (OliverEGreen) [reply src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs:377]: Done

File: src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs (hunk 2 above)
Comment (OliverEGreen) [reply src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs:360]: So annoyingly we can't really use `ElementState`, I've had to use `InfoBubbleViewModel.Style` instead, which is basically the same thing but it included `Info` as a state.

Comment (SHKnudsen) [reply src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs:360]: i see, was just looking at this whcih seems like we were using `State` to determine color

```csharp
/// <summary>
/// Sets the color of the warning bar, which informs the user that the node is in
/// either an error or a warning state.
/// </summary>
/// <param name="elementState"></param>
private void SetWarningBarColor(ElementState elementState)
{
    switch (elementState)
    {
        case ElementState.Warning:
            WarningBarColor = (SolidColorBrush)SharedDictionaryManager.DynamoColorsAndBrushesDictionary["NodeWarningColor"];
            break;
        case ElementState.Error:
            WarningBarColor = (SolidColorBrush)SharedDictionaryManager.DynamoColorsAndBrushesDictionary["NodeErrorColor"];
            break;
    }
}

Comment (SHKnudsen) [reply src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs:360]: oh seems like that has changed now 😁

Comment (SHKnudsen) [reply src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs:360]: Maybe see if possible to bind directly to `InfoBubbleViewModel.Style`

Comment (OliverEGreen) [reply src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs:360]: Got the zoom converter working ok, not having a ton of luck with binding to the `NodeViewModel.ErrorBubble.InfoBubbleStyle`

**Pull request closed without merging**